src/runtime/markdown-parser/handler/utils.ts

```diff
@@ -18,15 +18,19 @@ export function parseThematicBlock (lang: string) {
     }
   }
 
-  const language = lang.replace(/[{|[](.+)/, '').match(/^[^ \t]+(?=[ \t]|$)/)
-  const highlightTokens = lang.match(/{([^}]*)}/)
-  const filenameTokens = lang.match(/\[([^\]]*)\]/)
-  const meta = lang.replace(/^\w*\s*(\[[^\]]*\]|\{[^}]*\})?\s*(\[[^\]]*\]|\{[^}]*\})?\s*/, '')
+  const languageMatches = lang.replace(/[{|[](.+)/, '').match(/^[^ \t]+(?=[ \t]|$)/)
+  const highlightTokensMatches = lang.match(/{([^}]*)}/)
+  const filenameMatches = lang.match(/\[((\S)*)\]/) /** Allow ']' in filename. */
```

Comment:
I can see that you don't want an escape character to have `]` in your filename, but we didn't want to depend on spaces. If we now change this, it could be a breaking change.

My variant from before would introduce an escape character (e.g. `\\`) for `]` but
- would not need spaces after the last `]` that belongs to the filename
- allow spaces in the filename.


You could not do this anymore:

```md
```ts [filename]meta=[]
```

or

```md
```ts [filename/title with spaces]
```

Spaces in the filename for example are already used in the `nuxt/nuxt` docs. Some examples:
- https://github.com/nuxt/nuxt/blob/350788419c414317354de9900cd049fa750456cc/docs/7.migration/4.meta.md?plain=1#L27
- https://github.com/nuxt/nuxt/blob/350788419c414317354de9900cd049fa750456cc/docs/1.getting-started/5.routing.md?plain=1#L17
- https://github.com/nuxt/nuxt/blob/350788419c414317354de9900cd049fa750456cc/docs/1.getting-started/5.routing.md?plain=1#L25

---

#2067 was the last change to this regexes, there should also be some ideas.

Reply:
you've convinced me. I will update regex to allow esacape ] and update docs in that ways (documenting why we choose this method).